packages/neuron-wallet/tests/utils/export-history/index.test.ts

```diff
@@ -1,29 +1,147 @@
-import { getConnection } from 'typeorm'
+import os from 'os'
+import fs from 'fs'
+import path from 'path'
+import { initConnection, closeConnection, saveTransactions } from '../../setupAndTeardown/index'
+import transactions from '../../setupAndTeardown/transactions.fixture'
 import i18n from '../../../src/locales/i18n'
-import TransactionPersistor from '../../../src/services/tx/transaction-persistor'
-import { initConnection } from '../../../src/database/chain/ormconfig'
-import Transaction from '../../../src/models/chain/transaction'
-import fixtures from './fixtures.json'
-
-describe.skip('Test exporting history', () => {
-  // TODO: reuse test cases of TransactionsService#getAllByAddresses
-  beforeAll(async () => {
+import AddressService from '../../../src/services/addresses'
+import exportHistory from '../../../src/utils/export-history'
+
+describe('Test exporting history', () => {
+  const WALLET_ID = 'wallet id'
+  const MAINNET_CHAIN_TYPE = 'ckb'
+  const TESTNET_CHAIN_TYPE = 'ckb_testnet'
+  const FILE_PATH = path.join(os.tmpdir(), 'transaction.csv')
+  const stubProvider = {
+    walletID: WALLET_ID,
+    chainType: MAINNET_CHAIN_TYPE,
+    filePath: FILE_PATH,
+  }
+
+  beforeAll(() => {
     i18n.changeLanguage('en')
-    await initConnection(":memory:")
-    for (const fixture of Object.values(fixtures.transactions)) {
-      const tx = Transaction.fromSDK(
-        fixture.transaction as any,
-        fixture.transaction as any,
-      )
-      tx.outputs.forEach((o, i) => o.daoData = (fixture.transaction.outputs[i] as any).daoData)
-      await TransactionPersistor.saveFetchTx(tx)
+  })
+
+  beforeEach(async () => {
+    await initConnection()
+    return saveTransactions(transactions)
+  })
+
+  afterEach(() => {
+    try {
+      fs.unlinkSync(stubProvider.filePath)
+    } catch {
+      // ignore
     }
+    stubProvider.walletID = WALLET_ID
```

Comment:
The alternative way to do this is initializing the stubbed provider in the `beforeEach`, so you don't need to reset the data in `afterEach`, for example:
```
let stubProvider

  beforeAll(() => {
    stubProvider = {
      walletID: WALLET_ID,
      chainType: MAINNET_CHAIN_TYPE,
      filePath: FILE_PATH,
    }
  })
```

Reply:
Updated in https://github.com/nervosnetwork/neuron/pull/1738/commits/3ee84eca552ac3b7e467f3823b8c5a8b76766ac8